test/e2e.sh

```diff
@@ -0,0 +1,80 @@
+#! /bin/bash
+
+set -eux -o pipefail 
+
+# Get running RHCOS version. Assumes all nodes are running the same version.
+# TODO check RT kernel version
+get_rhcos_version() {
+    worker_node=$(kubectl get nodes --selector='node-role.kubernetes.io/worker' -ojsonpath='{.items[].metadata.name}')
+    rhcos_version=$(oc debug node/${worker_node} \
+	    --quiet \
+	    -- cat /host/etc/os-release \
+	    | grep "OSTREE_VERSION" \
+	    | cut -d= -f2 \
+	    | tr -d \'\")
+
+    echo ${rhcos_version}
+}
+
+# Check if driver-toolkit imagestream is available
+test_imagestream() {
+    oc get imagestream/driver-toolkit -n openshift
+}
+
+# Check for latest and RHCOS tags
+test_imagestream_tags() {
+    rhcos_version=$(get_rhcos_version)
+    echo "INFO: Node RHCOS version: ${rhcos_version}"
+
+    oc get imagestream/driver-toolkit -n openshift -o json | jq ".spec.tags[].name"
+    
+    image_latest=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"latest\") | .from.name")
+    echo "INFO: Image from latest tag: ${image_latest}"
+        
+    image_rhcos_tag=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"${rhcos_version}\") | .from.name")
+    echo "INFO: Image from RHCOS tag: ${image_rhcos_tag}"
+
+    [[ $image_latest != "" ]]
+
+    [[ $image_rhcos_tag != "" ]]
+    
+    [[ $image_rhcos_tag == $image_latest ]]
+}
+
+# Check that DTK contains the right kernel version
+test_kernel_version() {
+    node_kernel=$(oc get nodes -ojsonpath='{.items[].status.nodeInfo.kernelVersion}')
+    echo "INFO: Node kernel: ${node_kernel}"
+
+    dtk_release=$(oc debug --image-stream="openshift/driver-toolkit:latest" \
+	    --quiet \
+            -- cat /etc/driver-toolkit-release.json)
+    dtk_kernel=$(echo "${dtk_release}" | jq -r .KERNEL_VERSION)
+    echo "INFO: driver-toolkit kernel: ${dtk_kernel}"
+    
+    [[ ${dtk_kernel} ==  ${node_kernel} ]]
+}
+
+list_kernel_packages() {
+    # Check that DTK contains all the packages
+    oc debug --image-stream="openshift/driver-toolkit:latest" \
+            --quiet \
+            -- dnf list installed | grep kernel
+}
+
+
+printf "#### Driver Toolkit e2e test ####"
+
+printf "\n## TEST: Checking imagestream/driver-toolkit ##"
```

Comment:
why `printf` not `echo`?

(I would use `printf` only for very specific cases of unprintable characters, and `\n` isn't part of these special cases ;-))

besides:
```
## TEST: Checking that driver-toolkit/imagestream latest and RHCOS tag are matching and non-empty ##+ test_imagestream_tags
```
because `printf` doesn't implicitly add the final final `\n`

Reply:
Good catch. I was just trying to print newlines without extra echo statements, but the echos are cleaner.